ecosystem/node-checker/src/evaluators/metrics/consensus/round.rs

```diff
@@ -0,0 +1,141 @@
+use serde::{Deserialize, Serialize};
+
+// TODO: When we have it, switch to using a crate that unifies metric names.
+// As it is now, this metric name could change and we'd never catch it here
+// at compile time.
+const METRIC: &str = "aptos_consensus_last_committed_round";
+
+#[derive(Clone, Debug, Deserialize, Parser, PoemObject, Serialize)]
+pub struct ConsensusRoundEvaluatorArgs {}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct ConsensusRoundEvaluator {
+    args: ConsensusRoundEvaluatorArgs,
+}
+
+impl ConsensusRoundEvaluator {
+    pub fn new(args: ConsensusRoundEvaluatorArgs) -> Self {
+        Self { args }
+    }
+
+    fn get_consensus_round(
+        &self,
+        metrics: &PrometheusScrape,
+        metrics_round: &str,
+    ) -> GetMetricResult {
+        let evaluation_on_missing_fn = || {
+            self.build_evaluation_result(
+                "Consensus round metric missing".to_string(),
+                0,
+                format!(
+                    "The {} set of metrics from the target node is missing the metric: {}",
+                    metrics_round, METRIC
+                ),
+            )
+        };
+        get_metric(metrics, METRIC, None, evaluation_on_missing_fn)
+    }
+
+    #[allow(clippy::comparison_chain)]
+    fn build_evaluation(&self, previous_round: u64, latest_round: u64) -> EvaluationResult {
+        if latest_round < previous_round {
+            self.build_evaluation_result(
+                "Consensus round went backwards!".to_string(),
+                0,
+                format!("Successfully pulled metrics from target node twice, but the second time the consensus round went backwards (from {} to {}", previous_round, latest_round),
+            )
+        } else if latest_round == previous_round {
+            self.build_evaluation_result(
+                "Consensus round is not progressing".to_string(),
+                50,
+                "Successfully pulled metrics from target node twice, but the consensus round isn't progressing.".to_string(),
+            )
+        } else {
+            self.build_evaluation_result(
+                "Consensus round is increasing".to_string(),
+                100,
+                format!("Successfully pulled metrics from target node twice and saw that consensus round increased (from {} to {})", previous_round, latest_round),
+            )
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Evaluator for ConsensusRoundEvaluator {
+    type Input = MetricsEvaluatorInput;
+    type Error = MetricsEvaluatorError;
+
+    /// Assert that the consensus round is increasing on the target node.
+    async fn evaluate(&self, input: &Self::Input) -> Result<Vec<EvaluationResult>, Self::Error> {
+        let mut evaluation_results = vec![];
+
+        let previous_round = match self.get_consensus_round(&input.previous_target_metrics, "first")
+        {
+            GetMetricResult::Present(metric) => Some(metric),
```

Comment:
nit: it looks like we're starting to do this pattern a bunch (across evaluators). Might be worth creating a utils function.